blog/_posts/2021-03-15-scaling-inlets.md

```diff
@@ -0,0 +1,349 @@
+  Renewal Time:            2021-05-13T10:01:01Z
+  Revision:                1
+Events: 
+```
+
+Now everything is ready to create a first tunnel.
+
+
+## Creating the first exit-node
+
+Before we start our first tunnel, we need to know what variables or values of the Helm chart we have to configure.
+
+At the moment of writing, the Helm chart will have to following properties by default:
+
+- the default data plane port is set to 9090
+
+As not all applications are listening on port 9090, this has to be changed to the correct port.
+
+- the inlets token is read from the secret named `inlets-pro-secret`
+
+Here we can take two different approaches; either we create a single token that every tunnel will use, or we create a different token for each tunnel which will improve security as the same token should not need to be shared.
+For this tutorial, we chose the latter.
+
+- the ingress has cert-manager annotation to create a certificate automaticly with the HTTP01 challenge solver
+- the certifcate is available in a secret with a name like `<domain>-tls-secret`
+
+Because we have already a wildcard certificate available, we need to overwrite those settings as well. Otherwise cert-manager will still issue a certificate for the particular subdomain.
+
+
+Let's say the first tunnel we would like to create is for making our on premise [Keycloak](https://www.keycloak.org/) available from anywhere.
+
+Get the inlets-pro helm chart, generate a token for the inlets server and install the chart for the Keycload service:
+
+``` bash
+git clone https://github.com/inlets/inlets-pro
+
+kubectl create secret generic inlets-keycloak-token \
+  --from-literal token=$(head -c 16 /dev/random | shasum|cut -d" " -f1)
+
+helm upgrade --install keycloak ./inlets-pro/chart/inlets-pro \
+  --set tokenSecretName=inlets-keycloak-token \
+  --set ingress.annotations=null \
+  --set ingress.secretName=inlets-tls \
+  --set ingress.domain=keycloak-tunnel.inlets.example.com \
+  --set dataPlane.ports[0].port=8080 \
+  --set dataPlane.ports[0].targetPort=8080
+```
+
+What is the result of installing this chart:
+
+- a Pod with the inlets PRO server is running
+- a Control Plane service of type ClusterIP is created, exposing port 8123
+- an Ingress is created with the wildcard certificate, making the Control Plane service available in a secure manner
+- a Data Plane service of type ClusterIP is created, exposing port 8080
+
+This means that a inlets PRO client can connect to the Control Plane using the proper domain name, e.g. `wss://keycloak-tunnel.inlets.example.com`, and can punch out port 8080, making it accessible from only within this cluster, because of type ClusterIP.
+
+## Scaling inlets PRO tunnels
+
+Now that we have created a first tunnel, it is time to add all the other required tunnel.
+As demonstrated, the Helm chart is pretty convenient to get everything up and running and can be installed many times to create different tunnels.
```

Comment:
"is pretty convenient" -> "is more convenient than creating and managing individual virtual machines"